task/mongo-tasks.js

```diff
@@ -232,7 +750,51 @@ async function task_1_17(db) {
  *       https://docs.mongodb.com/manual/reference/operator/aggregation/dateFromString/
  */
 async function task_1_18(db) {
-    throw new Error("Not implemented");
+    let result = await db.collection('orders').aggregate([
+        {
+          '$project': {
+            '_id': 0, 
+            'OrderDate': {
+              '$dateFromString': {
+                'dateString': '$OrderDate'
+              }
+            }
+          }
+        }, {
+          '$match': {
+            '$expr': {
+              '$eq': [
+                {
+                  '$year': '$OrderDate'
+                }, 1998
```

Comment:
And you can use simple $eq here instead of `$expr`
NOTES: `expr` can have issues with indexes.

Reply:
If I remove $expr, how I can get $year value and pass to $eq for comparing? I can add a new 'year' field but whether it is necessary?

Reply:
You don't need filter by exact int value here. You can make filter with Date and simple interval comparison.

Reply:
Done :)